src/main/java/duke/storage/DukeFileWriter.java

```diff
@@ -0,0 +1,65 @@
+package duke.storage;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.ui.Ui;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Writes file input into internal storage
+ */
+public class DukeFileWriter {
+
+    /**
+     * Saves the task list into internal storage
+     * @param taskList the current instance of task list used by Duke
+     * @param filePath the file path of the internal storage to save at
+     */
+    public static void saveTaskListInInternalStorage(List<Task> taskList, String filePath) {
+        try {
+            String toBeSavedTaskListString = "";
+            for (int i = 0; i < taskList.size(); i++) {
+                String currentLine = "";
+                Task task = taskList.get(i);
+                if (task instanceof Event) {
```

Comment:
Is there a better way to do this instead of using instanceof? Maybe implement a checking system.

Reply:
Would there be a drawback in using `instanceof`? I implemented it this way because it was convenient.

Reply:
To be honest im not too sure, but I think we were taught in 2030 that the use of instanceof breaks the open/close principle in some way you may refer to this StackOverflow page: https://stackoverflow.com/questions/20589590/why-not-use-instanceof-operator-in-oop-design